File: c/claude/p121.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    // Allocate two chunks of memory, each of size 10
    char* chunk1 = (char*)malloc(10 * sizeof(char));
    char* chunk2 = (char*)malloc(10 * sizeof(char));
    
    // Initialize chunk1 with some data
    strcpy(chunk1, "HelloWorld");
    
    // Copy contents of chunk1 into chunk2
    strcpy(chunk2, chunk1);
    
    // Print contents of chunk2
    printf("Contents of chunk2: %s\n", chunk2);
    
    // Free memory allocated to chunk1
    free(chunk1);
    
    // Print contents of chunk2 (after freeing chunk1)
    printf("Contents of chunk2 after freeing chunk1: %s\n", chunk2);
    
    // Free memory allocated to chunk2
    free(chunk2);
    
    return 0;
}

File: java/gemini/p121.java
import java.util.Arrays;

public class p121 {
    public static void main(String[] args) {
        // Allocate two chunks
        int[] chunk1 = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
        int[] chunk2 = new int[10];

        // Copy contents
        System.arraycopy(chunk1, 0, chunk2, 0, 10);

        // Print contents of chunk2
        System.out.println("Chunk 2: " + Arrays.toString(chunk2));

        // "Free" chunk 1
        chunk1 = null; 
        System.out.println("Chunk 1 reference cleared.");

        // Print contents of chunk2
        System.out.println("Chunk 2: " + Arrays.toString(chunk2));

        // "Free" chunk 2
        chunk2 = null;
    }
}